Sources/Security/FoundationSecurity.swift

```diff
@@ -0,0 +1,101 @@
+//////////////////////////////////////////////////////////////////////////////////////////////////
+//
+//  FoundationSecurity.swift
+//  Starscream
+//
+//  Created by Dalton Cherry on 3/16/19.
+//  Copyright © 2019 Vluxe. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+//////////////////////////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import CommonCrypto
+
+public enum FoundationSecurityError: Error {
+    case invalidRequest
+}
+
+public class FoundationSecurity  {
+    var allowSelfSigned = false
+    
+    public init(allowSelfSigned: Bool = false) {
+        self.allowSelfSigned = allowSelfSigned
+    }
+    
+    
+}
+
+extension FoundationSecurity: CertificatePinning {
+    public func evaluateTrust(trust: SecTrust, domain: String?, completion: ((PinningState) -> ())) {
+        if allowSelfSigned {
+            completion(.success)
+            return
+        }
+        
+        if let validateDomain = domain {
+            SecTrustSetPolicies(trust, SecPolicyCreateSSL(true, validateDomain as NSString?))
+        }
+        
+        handleSecurityTrust(trust: trust, completion: completion)
+    }
+    
+    private func handleSecurityTrust(trust: SecTrust, completion: ((PinningState) -> ())) {
+        if #available(iOSApplicationExtension 12.0, *) {
```

Comment:
shouldn't this be `if #available(iOS 12.0, *)`?

Reply:
It should. Thanks for the heads up